src/IR.h

```diff
@@ -362,7 +362,7 @@ struct Evaluate : public StmtNode<Evaluate> {
 struct Call : public ExprNode<Call> {
     std::string name;
     std::vector<Expr> args;
-    typedef enum {Image, Extern, Halide, Intrinsic} CallType;
+    typedef enum {Image, Extern, PureExtern, Halide, Intrinsic, PureIntrinsic} CallType;
```

Comment:
Might be worth adding a comment here. In particular I'd like to record that the Pure is a property of the thing being called, not the call site. I.e. there should never be two calls to the same function, one of which is Pure and other not. (Though enforcing this for PureExtern is difficult.) I'm not sure this invariant is all that critical to optimization, but I believe it is the intent of the IR design and thus should be documented.